Passing additional surface variables into the atmospheric model

zhihong-tan opened the pull request:
This PR is linked to issue #4 and it depends on issue #92 (#93) of the FMScoupler repository.

A new boundary layer scheme (NCEP TKE-based eddy-diffusivity mass-flux scheme) is being implemented in the prototype-AM5 model, and it requires five additional surface variables (shflx, lhflx, wind, thv_atm, and thv_surf) as inputs. Thus, I made changes in the FMScoupler (please refer to its issues #92/#93), which would require adding a few lines in the atmos_null (atmos_model.F90) to include these variables in the land_ice_atmos_boundary_type structure.

Commit:
ADDED SURFACE VARIABLES REQUIRED BY NCEP TKE-EDMF

The shflx, lhflx, wind, thv_atm, thv_surf variables are passed
from the surface coupler to the atmospheric model as required by
the NCEP TKE-EDMF scheme.

rem1776 commented on atmos_model.F90:
```diff
@@ -737,6 +740,13 @@ subroutine lnd_ice_atm_bnd_type_chksum(id, timestep, bnd_type)
     write(outunit,100) 'lnd_ice_atm_bnd_type%q_star        ',mpp_chksum(bnd_type%q_star         )
+#ifndef use_AM3_physics
+    write(outunit,100) 'lnd_ice_atm_bnd_type%shflx         ',mpp_chksum(bnd_type%shflx          )!miz
+    write(outunit,100) 'lnd_ice_atm_bnd_type%lhflx         ',mpp_chksum(bnd_type%lhflx          )!miz
```
This is causing a segfault in the FMScoupler "null_model_build.sh" test script. I think putting an `if(associated(bnd_type%shflx)` and `if(associated(bnd_type%lhflx)` in front of the respective `write`'s should fix it.

Commit:
ADDED IF-STATEMENTS FOR CHECKSUM

Two if-associated statements are added before the checksums
of bnd_type%shflx and bnd_type%lhflx to pass the null model
build.
